docs/designs/supporting_pytorch.md

```diff
@@ -0,0 +1,316 @@
+# Supporting PyTorch
+
+## Introduction
+
+ElasticDL is an open-source distributed deep learning programming framework based
+on TensorFlow and Kubernetes. Considering that PyTorch is more widely used in academia,
+this project will support PyTorch.
+
+## How PyTorch is Used
```

Comment:
Suggestion: Instead of splitting one PyTorch program into pieces, you may want to have one PyTorch program and then explains each part of it. Btw, the point of this section should be, showing current PyTorch usage (PyTorch APIs), so that readers have some sense of how easy or not easy that PyTorch is being used.

With the next sections of EDL design (to supports PyTorch), which should be the main purpose of this doc, updated (maybe or maybe not) ElasticDL APIs (when working with PyTorch) should be designed and shown, so that readers understand your design points and how easy it is to use EDL when training a PyTorch model.

Reply:
> Suggestion: Instead of splitting one PyTorch program into pieces, you may want to have one PyTorch program and then explains each part of it. Btw, the point of this section should be, showing current PyTorch usage (PyTorch APIs), so that readers have some sense of how easy or not easy that PyTorch is being used.
> 
> With the next sections of EDL design (to supports PyTorch), which should be the main purpose of this doc, updated (maybe or maybe not) ElasticDL APIs (when working with PyTorch) should be designed and shown, so that readers understand your design points and how easy it is to use EDL when training a PyTorch model.

Your suggestion is very helpful and I will improve the documentation.